agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -455,26 +466,42 @@ func (cs *CredentialSpecResource) writeS3File(writeFunc func(file oswrapper.File
 	if err != nil {
 		return err
 	}
-	defer temp.Close()
 
 	err = writeFunc(temp)
 	if err != nil {
 		return err
 	}
 
-	err = temp.Chmod(os.FileMode(filePerm))
+	// Persist the file to disk.
+	err = temp.Sync()
 	if err != nil {
 		return err
 	}
 
-	// Persist the file to disk.
-	err = temp.Sync()
+	input, err := ioutil.ReadFile(temp.Name())
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filePath, input, 0644)
```

Comment:
Why the permission is 0644 here instead of **filePerm**?

Reply:
Why do we need 755 permissions. 0644 should be enough and it has to be the same the other place as well. I changed the filePerm to 0644